### PR TITLE
rsync: switch to openssl 3.1

### PR DESCRIPTION
**Pull request by iigs**

### CODE REVIEW

File: components/network/rsync/Makefile [hunk 1]
```diff
@@ -1,22 +1,12 @@
 #
-# CDDL HEADER START
+# This file and its contents are supplied under the terms of the
```
Comment (mtelka) [inline components/network/rsync/Makefile]: Is it okay to change the copyright notice of already existing file with the Oracle copyright in it?

Comment (AndWac) [reply components/network/rsync/Makefile]: The original text references http://www.opensolaris.org/os/licensing which is not available anymore. The new text is the best replacement as it seems to be in the same vein and continues to use the CDDL.

**Comment by iigs** [comment]
I found at least one example of an open file from Oracle with an original CDDL header, last modified post merge by Oracle.  The file contains an Oracle copyright, which an Illumos dev did not change the header when added for Illumos use.  Even  though this is slightly different use case, I will follow that example for header use here.

**Comment by mtelka** [comment]
> I found at least one example of an open file from Oracle with an original CDDL header, last modified post merge by Oracle.

URL?  Thank you.

**Pull request merged**